src/main/java/Azir/Azir.java

```diff
@@ -0,0 +1,149 @@
+                String[] result = Parser.parse(fullCommand, tasks.getSize());
+                switch (result[0]) {
+                case "list":
+                    ui.showCommandEndMessage(fullCommand);
+                    for (int i = 0; i < tasks.getSize(); i++) {
+                        System.out.printf("%d. %s\n", i + 1, tasks.getTask(i));
+                    }
+                    break;
+                case "mark":
+                    Task chosenTask = tasks.getTask(Integer.valueOf(result[1]) - 1);
+                    chosenTask.setDone();
+                    ui.showCommandEndMessage(result[0], chosenTask.toString());
+                    break;
+                case "unmark":
+                    Task task = tasks.getTask(Integer.valueOf(result[1]) - 1);
+                    task.setNotDone();
+                    ui.showCommandEndMessage(result[0], task.toString());
+                    break;
+
+                case "delete":
+                    Task deletedTask = tasks.getTask(Integer.valueOf(result[1]) - 1);
+                    tasks.DeleteTask(Integer.valueOf(result[1]) - 1);
+                    ui.showCommandEndMessage(result[0], deletedTask.toString());
+                    ui.showTaskNumber(tasks.getSize());
+                    break;
+
+                case "todo":
+                    if (fullCommand.length() == 4) {
+                        throw new AzirException("todo cannot have an empty description. " +
+                                "Format: todo [description]");
+                    }
+                    Task todoTask = new Todo(result[1]);
+                    tasks.addTask(todoTask);
+                    ui.showCommandEndMessage(result[0], todoTask.toString());
+                    ui.showTaskNumber(tasks.getSize());
+                    break;
+
+                case "deadline":
+                    String dateFormat = "yyyy-MM-dd";
+                    DateTimeFormatter formatter = DateTimeFormatter.ofPattern(dateFormat);
+                    try {
+                        Task deadlineTask = new Deadline(result[1], LocalDate.parse(result[2], formatter));
+                        tasks.addTask(deadlineTask);
+                        ui.showCommandEndMessage(result[0], deadlineTask.toString());
+                        ui.showTaskNumber(tasks.getSize());
+                    } catch (DateTimeParseException e) {
+                        throw new AzirException("deadline needs to be in the following format: yyyy-mm-dd");
+                    }
+                    break;
+
+                case "event":
+                    Task eventTask = new Event(result[1], result[2], result[3]);
+                    tasks.addTask(eventTask);
+                    ui.showCommandEndMessage(result[0], eventTask.toString());
+                    ui.showTaskNumber(tasks.getSize());
+                    break;
+
+                case "find":
+                    ui.showCommandEndMessage(result[0]);
```

Comment:
I feel the name 'showCommandEndMessage' is a bit not self-explanatory. Maybe 'replyListQuery' or other equivalent would be better?

Reply:
Good idea. Thanks